src/Program.ts

```diff
@@ -60,7 +60,7 @@ export class Program {
         /**
          * The root directory for this program
          */
-        public options: BsConfig,
+        public options: FinalizedBsConfig,
```

Comment:
If you'd like this to be a backwards-compatible change we could let `BsConfig` be passed here, and then apply `normalizeConfig` inside of the constructor. This feels a little weird to me, because it's letting potentially invalid data get pretty far into the program before getting normalized, but I can't think of any concrete cases where it would be harmful.

Reply:
Yeah. Let's just declare `options` as a field down below, and let the incoming type be `BsConfig`. We're already doing a `normalizeConfig` in the constructor anyway.
image.png

Reply:
Let's keep this PR as-is, and open new ones with other stuff. Feel free to open those other PRs targeting this one so you don't have to keep remembering which one is next. They'll get back in sync as we merge.